### PR TITLE
Getting started: solving kata

### PR DESCRIPTION
**Pull request by hobovsky**
Please take a look if it goes in good direction. I am wondering if it needs screenshots, or links to UI docs will suffice?

### CODE REVIEW

**Comment by kazk** [comment]
Ugh, the output of TextLint is not helpful. I'll look for a way to integrate with GitHub Actions better, but you have a misspelled word on line 23.

**Comment by hobovsky** [comment]
Yeah it's kinda confusing, but until now I somehow managed to understand it. I will fix the typo tomorrow

**Comment by Blind4Basics** [comment]
there was still no preview available, so I made a small change/commit.

File: content/getting-started/solving-kata.md [hunk 1]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+When you unlock solutions you immediately gain access to all information about kata just as if you have finished it: solutions of other users, code of full test suite, etc. Although be warned, because this way you get no reward, and you forfeit all eligibility for any reward you would get for solving this kata in the future, for example in another language. Use this option with care, giving up is also a way to learn on Codewars, but sometimes waiting a bit and doing more research also helps.
+
+## Rewards
+
+You worked hard to solve your last challenge, and after you succeeded, you definitely deserve a reward. Each solved tasks rewards you in two ways: progressing your rank, and granting some amount of Honor points. The more difficult kata you solve, the bigger the reward.
```
Comment (Blind4Basics) [inline content/getting-started/solving-kata.md:51]: linefeed

```suggestion
You worked hard to solve your last challenge, and after you succeeded, you definitely deserve a reward.

Each solved tasks rewards you in two ways: progressing your rank, and granting some amount of Honor points. The more difficult kata you solve, the bigger the reward.
```

File: content/getting-started/solving-kata.md [hunk 2]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+## Unlocking solutions
+
+Some kata are difficult. Some of them turn out to be more difficult than they initially appeared. Sometimes it might just not be your good day. It's OK, and happens to all of us. You can take a break from the kata and try another one, you will always be able to come back to this one later. You can do more research, search the web for algorithms, or ask friends for help. But if nothing helps, and you _really_ need to know what other users came up with to defeat the task, you can unlock solutions without finishing the kata.
+
+When you unlock solutions you immediately gain access to all information about kata just as if you have finished it: solutions of other users, code of full test suite, etc. Although be warned, because this way you get no reward, and you forfeit all eligibility for any reward you would get for solving this kata in the future, for example in another language. Use this option with care, giving up is also a way to learn on Codewars, but sometimes waiting a bit and doing more research also helps.
+
+## Rewards
```
Comment (Blind4Basics) [inline content/getting-started/solving-kata.md:49]: about the rewards, I know you're a "rank" guy, but most users will be focused on honor first. So I'd rather begin with honor points.

File: content/getting-started/solving-kata.md [hunk 3]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+You worked hard to solve your last challenge, and after you succeeded, you definitely deserve a reward. Each solved tasks rewards you in two ways: progressing your rank, and granting some amount of Honor points. The more difficult kata you solve, the bigger the reward.
+
+### Rank
+
+Rank level reflects your experience and knowledge you have gained by solving kata. You start at the lowest rank of 8 kyū, and each correct solution brings you closer to a higher rank. You can see your rank progress toward next level on your profile page:
```
Comment (Blind4Basics) [inline content/getting-started/solving-kata.md:55]: LF + other

```suggestion
Rank level reflects the experience and knowledge you gained by solving kata.

You start at the lowest rank of 8 kyū, and each correct solution brings you closer to a higher rank. You can see your rank progress toward next level on your profile page:
```

File: content/getting-started/solving-kata.md [hunk 4]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+![rank progress](./img/solving_01_rank-progress.png)
+
+More information on ranks and progress can be found [here (TODO: add link to docs on progress)]().
+
+### Honor
```
Comment (Blind4Basics) [inline content/getting-started/solving-kata.md:61]: you so much don't like honor that you didn't even show the top right corner of the page to show where the info is visible to the user while you showed the progress wheel... XD

;)

=> add a picture?

File: content/getting-started/solving-kata.md [hunk 5]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+Honor points are rewarded by contributing to Codewars in many ways, and solving a kata is one of them. By earning Honor points you gain additional [privileges (TODO: link to docs)]() and climb [leaderboards (TODO: link to docs)]().
+
+## Feedback
```
Comment (Blind4Basics) [inline content/getting-started/solving-kata.md:65]: might be good to be a bit more specific about the feedback. It's actually not about "did you lke it or not", but rather "was the kata up to quality standards" (since that's what we do/use during the beta process)

File: content/getting-started/solving-kata.md (hunk 4 above)
Comment (hobovsky) [reply content/getting-started/solving-kata.md:61]: I did not want to put too many pictures here, because I think it's easily visible even for newbies, and that actual details will be present in "UI docs", possibly linked from here

File: content/getting-started/solving-kata.md (hunk 2 above)
Comment (hobovsky) [reply content/getting-started/solving-kata.md:49]: It's not that I am a "rank guy". It's more. I want to make all newbies "rank people". Drive them away from Honor as much as possible in early beginnings, and focus them on important things ;) Like knowledge.

File: content/getting-started/solving-kata.md [hunk 6]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+When sample tests pass and you believe your solution is ready, you can go to next step.
+
+### Attempt
+
+This button will submit your code to test it against the full set of tests needed to complete the kata.
+
+You cannot see full tests until you solve the kata. Full test suite is usually much larger than sample tests, and it contains test cases for various inputs, data configurations, edge cases, performance, etc. Even if your solution passes all sample tests, it still can fail attempt against full tests. If tests output panel turns red, do not worry! You can edit and run your solution as many times as you need.
+
+When tests output panel turns green and your solution passes all the tests, congratulations! Task is solved, and you are awarded with Honor points and your progress increases (see [following paragraph](#rewards) for more details).
```
[the remark this answers is not in the document]
Comment (hobovsky) [reply content/getting-started/solving-kata.md:33]: Yup, I was wondering about this too. I decided to describe it the way it is, and not the way we would like it to be.

File: content/getting-started/solving-kata.md [hunk 7]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+### Solution editor
+
+You use solution editor to write code which solves the task. You can type the code straight into the editor, but if you want, you can use your favorite IDE and just copy your solution from it and paste it here.
+
+### Test
+
+This button will test your solution against the _sample_ tests. Most kata will have provided you some sample tests to get you going, while others will not, in which case some test documentation will be shown instead.
+
+Codewars promotes approach of [Test Driven Development (is link to Wikipedia article OK?)](). This means that you are encouraged to write not only code for your solution, but also add as many tests in `Sample Tests` panel as you can think of (see [Writing Tests (TODO: add link to explanation how to add more sample tests)]() to see how), for various scenarios, inputs, and edge cases. Every time you run them by clicking `TEST`, system will execute all sample tests against your solution, and tell you if it passed all of them, or maybe failed some.
```
[the remark this answers is not in the document]
Comment (hobovsky) [reply content/getting-started/solving-kata.md:23]: Not everyone treats Wikipedia as a good source for domain-specific things. Maybe someone would like more "professional" explanation

File: content/getting-started/solving-kata.md [hunk 8]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
+
+### Attempt
+
+This button will submit your code to test it against the full set of tests needed to complete the kata.
+
+You cannot see full tests until you solve the kata. Full test suite is usually much larger than sample tests, and it contains test cases for various inputs, data configurations, edge cases, performance, etc. Even if your solution passes all sample tests, it still can fail attempt against full tests. If tests output panel turns red, do not worry! You can edit and run your solution as many times as you need.
+
+When tests output panel turns green and your solution passes all the tests, congratulations! Task is solved, and you are awarded with Honor points and your progress increases (see [following paragraph](#rewards) for more details).
+
+### Submit Final: Production Quality Code
```
Comment (kazk) [inline content/getting-started/solving-kata.md:35]:
```suggestion
### Submit Final
```

File: content/getting-started/solving-kata.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (kazk) [reply content/getting-started/solving-kata.md:23]: I think it's OK to link to Wikipedia. But having a page summarizing the basics will be nice to have in the future.

File: content/getting-started/solving-kata.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (kazk) [reply content/getting-started/solving-kata.md:33]: I'd like to change it and I'll try to find a way to undo the rewards given to those without submitting. See https://github.com/codewars/codewars.com/issues/1378#issuecomment-637087348

I don't have strong opinion about where to put it right now. We can always update the docs.

**Comment by hobovsky** [comment]
I split "solving" and "solved" parts into two separate pages. Please let me know what you think.

**Comment by Blind4Basics** [comment]
at the root, the `gridsome.config.js` file needs to be update (for the sidebar) :

```code
...
{
        name: "getting-started",
        sections: [
          {
            title: "Getting Started",
            items: [
              "/getting-started/",
              "/getting-started/registering/",
              "/getting-started/setting-up/",
              "/getting-started/finding-kata/",
              "/getting-started/solving-kata/",
              "/getting-started/kata-solved/",
              "/getting-started/solutions/",
              "/getting-started/community/",
            ],
          },
        ],
      },
...
```

**Comment by hobovsky** [comment]
> * at the root, the `gridsome.config.js` file needs to be update (for the sidebar) :

Done

> * It seems there are images at different levels in the folder (07 is outside `getting-started/img/`) =>?

It got submitted accidentally, sorry. It's to be removed.

> * would be better to update the images right now with dark/light versions, no? (see #74)

Does it work already?

**Comment by Blind4Basics** [comment]
> Does it work already?

actually yes. I forgot to update the title of the section... XD (corrected)

**Comment by Blind4Basics** [comment]
Maybe change the title of the new page to "When kata get solved"? (because other titles are plural, that looks a bit weird in the sidebar)

**Comment by hobovsky** [comment]
> Maybe change the title of the new page to "When kata get solved"? (because other titles are plural, that looks a bit weird in the sidebar)

I tried to fix capitalization, maybe it looks better now.

> would be better to update the images right now with dark/light versions, no? (see #74)

Created #86

**Comment by Blind4Basics** [comment]
> I tried to fix capitalization, maybe it looks better now.

:+1: 
that was it, actually :)

File: content/getting-started/solving-kata.md [hunk 9]
```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
...
--->
+After clicking on a link to a kata, you are presented with the [kata details (TODO: link to kata details panel doc)]() view with general information about it. Read carefully through the description, and if you are ready to face the challenge, click `TRAIN`.
+
+## Kata Trainer
+
+Welcome to the kata trainer! You use this view to take your attempts on solution, write the code and run tests. A few tips to get you started:
+
+### Solution editor
+
+You use solution editor to write code which solves the task. You can type the code straight into the editor, but if you want, you can use your favorite IDE and just copy your solution from it and paste it here.
```
[the remark this answers is not in the document]
Comment (hobovsky) [reply content/getting-started/solving-kata.md:18]: > About "when the tests fail over again" and "unlocking solutions":
> 
> * unlocking might lay into the current page because of the related button. But then I'd put it after the "submit final" section. Note: `## Unlocking solutions`, in this case, not `# Un...`
> * the trouble shooting part doesn't really feel right in this page. It's related to solving, yeah, for sure. I feel like if it's the last section of the page it's ok (then: submit final -> unlock -> trouble shooting -> end)

I was trying to follow the "natural" flow of the way how newbie would go through a kata:
1.  write solution,
2. Run sample tests
3. Run full tests, which potentially fail many times,
4. Go search for help,
5. Unlock solutions when nothing helps 

Order of headers follows this process, which I think is natural and different order seems a bit off to me.

**Comment by kazk** [comment]
@hobovsky You can merge this if you want. I already left my review yesterday. https://github.com/codewars/docs/pull/72#pullrequestreview-453634748
We can improve this after merging because all the comments are becoming difficult to keep track.

**Comment by hobovsky** [comment]
I've already started working on your suggestions, so if you'd want to review my updates, then go ahead, and if you think it;s ok, you can merge it. I will get back to it tomorrow, so if you still think things are good, I will merge my changes. But you can still leave some remarks if you have any, or just merge if you think all is ok.

File: content/getting-started/solving-kata.md [hunk 10]
```diff
@@ -1,15 +1,43 @@
...
+
+### Testing your solution with sample tests
+
+Codewars promotes the [Test Driven Development](https://en.wikipedia.org/wiki/Test-driven_development) approach. This means that you are encouraged to write not only code for your solution, but also add as many tests in `Sample Tests` panel as you can think of (see [Writing Tests (TODO: add link to explanation how to add more sample tests)]() to see how), for various scenarios, inputs, and edge cases. Most kata will have provided you some sample tests to get you going, while others will not, in which case some test documentation will be shown instead.
+Every time you run sample tests by clicking `TEST`, system will execute them against your solution, and tell you if it passed all of them, or maybe failed some. When some sample test fails, you know you need to work on your solution some more. When all sample tests pass, you can either add some more test cases for scenarios not tested yet, or, if you believe your solution is ready, you can go to next step.
+
+### Attempt your solution with full test suite
```
Comment (kazk) [inline content/getting-started/solving-kata.md:25]:
```suggestion
### Attempting to pass the full test suite
```

File: content/getting-started/solving-kata.md [hunk 11]
```diff
@@ -1,15 +1,43 @@
...
+## Kata Trainer
+
+Welcome to the kata trainer! You use this view to take your attempts on solution, write the code and run tests. A few tips to get you started:
+![kata trainer](./img/solving_03_trainer.png)
+
+### Solution editor
+
+You use solution editor to write code which solves the task. You can type the code straight into the editor, but if you want, you can use your favorite IDE and just copy your solution from it and paste it here.
+
+### Testing your solution with sample tests
```
Comment (kazk) [inline content/getting-started/solving-kata.md:20]:
```suggestion
### Testing with sample tests
```

File: content/getting-started/solving-kata.md [hunk 12]
```diff
@@ -1,15 +1,43 @@
...
+
+### Attempt your solution with full test suite
+
+When your solution is thoroughly tested, it passes all sample tests you created, and you cannot think of any new test cases to add, you can run full test suite against your solution. Passing it is necessary to have the kata marked as solved.
+
+You cannot see full tests until you solve the kata. Full test suite is usually much larger than sample tests, and it contains test cases for various inputs, data configurations, edge cases, performance, etc. Even if your solution passes all sample tests, it still can fail attempt against full tests. If tests output panel turns red, do not worry! You can edit and run your solution as many times as you need.
+
+When tests output panel turns green and your solution passes all the tests, congratulations! Task is solved, and you are awarded with Honor points and your progress increases (more on rewards on the next page).
+
+### Submitting Final Version of your Solution
```
Comment (kazk) [inline content/getting-started/solving-kata.md:33]:
```suggestion
### Submitting the final version
```
